extension/src/tests/mod.rs

```diff
@@ -0,0 +1,120 @@
+use crate::mock::RuntimeResult;
+use crate::tests::utils::decoding_failed_weight;
+use crate::{
+	environment::Ext,
+	mock::{
+		new_test_ext, Config, DispatchExtFuncId, MockEnvironment, MockExt, NoopFuncId,
+		ReadExtFuncId, RuntimeCall, RuntimeRead, Test,
+	},
+	Environment, Extension,
+};
+use codec::Encode;
+use frame_system::Call;
+use pallet_contracts::chain_extension::RetVal::Converging;
+pub(crate) use utils::{
+	charge_weight_filtering_dispatch_call, charge_weight_filtering_read_state,
+	extension_call_dispatch_call_weight, extension_call_read_state_weight,
+	function_dispatch_call_weight, function_read_state_weight, overhead_weight,
+	read_from_buffer_weight,
+};
+
+mod contract;
+mod utils;
+
+#[test]
+fn extension_call_works() {
+	let input = vec![2, 2];
+	let mut env = MockEnvironment::new(NoopFuncId::get(), input.clone(), MockExt::default());
+	let mut extension = Extension::<Config>::default();
+	assert!(matches!(extension.call(&mut env), Ok(Converging(0))));
+	assert_eq!(env.charged(), overhead_weight(input.len() as u32))
+}
+
+#[test]
+fn extension_for_unknown_function_works() {
```

Comment:
Change postfix to `_fails` please. If the weight is the thing that works, please adjust the name to something like `charge_weight_for_unknown_function_works"

Reply:
Agree with the postfix, but might be more useful to start tests that logically belong to the same component with the same prefix, so remain somewhat together when sorted. It might be more useful to see all tests for a function grouped together before then the test case and result in the test name.